### DISCUSSION TITLE
Home Assistant Media Library size limitations?

### DISCUSSION THREAD

**Discussion by oywino** [music-assistant/support#1410] [Q&A] [answered]
HI, my question may not be directly relevant for Music Assistant, but I'd like to check this out before installing.
I'm running HA Core v2023.7.3 in a docker container, In my docker compose YAML file I have the following media library set up:

`- /share/Music:/media:ro`

This seems to work, but only just. My media library contains over 10k albums. If try to browse the library using stock HA as-is, it is only able to display the first 1000 albums in an alphabetically sorted list when I use the `Album `button. This is how the end of the list looks like:

![image](https://github.com/music-assistant/hass-music-assistant/assets/18447119/6be9987c-2e3e-4e97-be4f-b2668583b9c4)

But if I use the `My media` button, then it gives me a list of all my albums, but only as grey folder icons with the album titles underneath.

I have tried several time to inquire the HA dev team members as to why there seems to be a limitation of 1K entries in the "usable" Albums list that properly displays AlbumArt, but to this day I have never been able to achieve as much as a single comment or feedback from anyone.
I thought perhaps you guys knew something about this, or if you have the same limitations in your integration?

**Comment by OzGav** [comment]
10000 albums must have cost you a lot of money?!

**Reply by oywino** [reply]
Indeed - but I've been doing this over a period of almost 40 years!
We all have hobbies - don't we? Mine hasn't been very costly.
Still, I'm curious as to why HA fails to display more than 1000 (one/tenth) of my album collection.

**Reply by marcelveldt** [reply]
The HA Media Browser is really not suitable for that amount of items. I strongly suggest you to use the MA interface to browse such a large collection as it is optimized with server-side paging etc. Loading more than 1000 items at once in the HA media browser will crash your browser or overload your HA core, its simply not designed for that. Also browsing it in that way is not very convenient, are going to scroll until your fingers bleed ? Better make use of filtering, sorting and search, all that is missing in the HA Media browser and included in the MA library. Just my 2 cents

**Reply by oywino** [reply]
Well, usability wasn't the issue. You may be right about that, but still I wonder if the 1k limit is by design or a bug?
Anyway, as soon as MA is out of beta, I will give it a go.

**Reply by marcelveldt** [reply, marked as answer]
Its by design for reasons I explained above.
